Mage.Sets/src/mage/cards/c/ContestOfClaws.java

```diff
@@ -0,0 +1,87 @@
+package mage.cards.c;
+
+import java.util.UUID;
+
+import mage.abilities.Ability;
+import mage.abilities.common.delayed.ReflexiveTriggeredAbility;
+import mage.abilities.effects.OneShotEffect;
+import mage.abilities.effects.keyword.DiscoverEffect;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.constants.CardType;
+import mage.constants.Outcome;
+import mage.filter.StaticFilters;
+import mage.game.Game;
+import mage.game.permanent.Permanent;
+import mage.target.common.TargetControlledCreaturePermanent;
+import mage.target.common.TargetCreaturePermanent;
+
+/**
+ *
+ * @author skaspels
+ */
+public final class ContestOfClaws extends CardImpl {
+
+    public ContestOfClaws(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.SORCERY}, "{1}{G}");
+        
+
+        // Target creature you control deals damage equal to its power to another target creature.
+        this.getSpellAbility().addEffect(new ContestofClawsEffect());
+        this.getSpellAbility().addTarget(new TargetControlledCreaturePermanent());
+        this.getSpellAbility().addTarget(new TargetCreaturePermanent(StaticFilters.FILTER_CREATURE_YOU_DONT_CONTROL));
+
+        // If excess damage was dealt this way, discover X, where X is that excess damage.
+
+    }
+
+    private ContestOfClaws(final ContestOfClaws card) {
+        super(card);
+    }
+
+    @Override
+    public ContestOfClaws copy() {
+        return new ContestOfClaws(this);
+    }
+}
+
+class ContestofClawsEffect extends OneShotEffect {
+    ContestofClawsEffect() {
+        super(Outcome.Benefit);
+        staticText = "Target creature you control deals damage equal to its power to another " +
+                "target creature. If excess damage was dealt this way, discover X, where X is that excess damage.";
+    }
+
+    private ContestofClawsEffect(final ContestofClawsEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public ContestofClawsEffect copy() {
+        return new ContestofClawsEffect(this);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        if (source.getTargets().size() != 2) {
+            throw new IllegalStateException("It must have two targets, but found " + source.getTargets().size());
+        }
+        Permanent myPermanent = game.getPermanent(getTargetPointer().getFirst(game, source));
+        if (myPermanent == null) {
+            return false;
+        }
+        int damageValue = myPermanent.getPower().getValue();
+        Permanent anotherPermanent = game.getPermanent(source.getTargets().get(1).getFirstTarget());
+        if (anotherPermanent == null){
+            return false;
+        }
+        int lethal = Math.min(anotherPermanent.getLethalDamage(source.getSourceId(), game), damageValue);
+        anotherPermanent.damage(damageValue, source.getSourceId(), source, game);
```

Comment:
this looks wrong. the source of the damage is the first target creature, not the spell.